python/ray/data/read_api.py

```diff
@@ -2842,8 +2842,20 @@ def from_huggingface(
 
     if isinstance(dataset, datasets.IterableDataset):
         # For an IterableDataset, we can use a streaming implementation to read data.
-        return read_datasource(HuggingFaceDatasource(dataset=dataset))
+        return read_datasource(
+            HuggingFaceDatasource(dataset=dataset),
+            parallelism=parallelism,
+            concurrency=concurrency,
+            override_num_blocks=override_num_blocks,
+        )
     if isinstance(dataset, datasets.Dataset):
+        # For non-streaming Hugging Face Dataset, we don't support override_num_blocks
+        if override_num_blocks is not None:
+            raise ValueError(
+                f"Invalid value for 'override_num_blocks': {override_num_blocks}. "
+                "Valid value is None for non-streaming Hugging Face Dataset"
```

Comment:
nit; adding a suggestion to resolve the error.
```suggestion
                f"`override_num_blocks` parameter is not supported for streaming Hugging Face Datasets. "
                f"Please omit the parameter or use non-streaming mode to read the dataset."
```

Reply:
Thanks for the suggestion, @scottjlee  just fixed it!

Another quick question here, since we only support following dataset as source
```python
def 
    ...
    dataset: Union["datasets.Dataset", "datasets.IterableDataset"],
   ...
```

do you think it's good if we can remove following code in same function? 
```python
    ...
    elif isinstance(dataset, (datasets.DatasetDict, datasets.IterableDatasetDict)):
        available_keys = list(dataset.keys())
        raise DeprecationWarning(
            "You provided a Hugging Face DatasetDict or IterableDatasetDict, "
            "which contains multiple datasets, but `from_huggingface` now "
            "only accepts a single Hugging Face Dataset. To convert just "
            "a single Hugging Face Dataset to a Ray Dataset, specify a split. "
            "For example, `ray.data.from_huggingface(my_dataset_dictionary"
            f"['{available_keys[0]}'])`. "
            f"Available splits are {available_keys}."
        )

```

Reply:
i think we should still keep it, even though the type hints give the information. we left this explicit type check since we found that users would often try to pass the initial Hugging Face dataset, which may contain multiple sub-datasets.